FileWriteFlags.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+namespace FileWriteFlag
+{
+	// Default values are Do not overwrite files, append to existing files, and write in binary format.
+	enum FileWriteFlag : unsigned int
+	{
+		// SET:     Allow overwritting the file if it exists (mutually exclusive from Append).
```

Comment:
I'm not sure these are really mutually exclusive. Append just sets the current position to the end of the file. This might not relate to read/write permissions, though usually you'd expect write permission in that case.

Reply:
I suppose you could manually move the position to the beginning of the file and start overwriting the file after selecting Append. I'm not sure why this would be done in practice though? I can remove the mutually exclusive part from the comment.